### DISCUSSION TITLE
Compute effective locking state the same as USD ComputeVisibility ?

### DISCUSSION THREAD

**Discussion by csyshing** [Autodesk/maya-usd#2825] [Q&A]
Hi,

Maya USD has the ability to lock an individual prim's xformOp (right click on the xformOp attribute and "Lock Attribute"), in our studio when using AL_USDMaya, we have the similar functions in [ProxyShape](https://github.com/Autodesk/maya-usd/blob/v0.21.0/plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h#L735) that computes effective locking state (recursively) in the same way as USD [ComputeVisibility()](https://github.com/PixarAnimationStudios/USD/blob/v22.11/pxr/usd/usdGeom/imageable.h#L326-L344), that if a parent prim is locked, the effective locking state of all children prims are also considered 'locked' even there is no explicit setting on individual children.

Would that be an option in Maya USD also have similar feature? An additional free form (recursive) function in [mayaUsd/utils/editability.h](https://github.com/Autodesk/maya-usd/blob/v0.21.0/lib/mayaUsd/utils/editability.h#L30)? And possibly extend [isAttributeEditAllowed()](https://github.com/Autodesk/maya-usd/blob/v0.21.0/lib/mayaUsd/ufe/Utils.cpp#L711) to include checking parent prims?

Thanks,
Zhicheng

**Comment by womanyoyoyo-adsk** [comment]
Hi @csyshing,

Are you able to explain the workflow that this is needed for? Is this for artists? We have different ways to achieve this with Maya data (Display layers, etc.) and I'm wondering if we can map it to an existing Maya pattern.

Natalia
